pydantic_ai_slim/pydantic_ai/embeddings/settings.py

```diff
@@ -0,0 +1,49 @@
+from typing import Literal, TypedDict
+
+
+class EmbeddingSettings(TypedDict, total=False):
+    # TODO: May want to add extra_headres, extra_query, extra_body, timeout, etc.
+
+    output_dimension: int
+    """The maximum number of tokens to generate before stopping.
+
+    Supported by:
+
+    * Cohere
+    * OpenAI
+    """
+
+    max_tokens: int
+    """The maximum number of tokens to generate before stopping.
+
+    Supported by:
+
+    * Cohere
+    """
+
+    # We don't support embedding_types for now because it doesn't affect the user-facing API today..
+    # embedding_types: Literal["float", "int8", "uint8", "binary", "ubinary", "base64"]
+
+    input_type: Literal['search_document', 'search_query', 'classification', 'clustering', 'image']
+    """The input type of the embedding.
+
+    Supported by:
+
+    * Cohere (See `cohere.EmbedInputType`)
```

Comment:
Following the pattern in `ModelSettings`, we should move any options only supported by one model to `{Cohere}EmbeddingSettings` with a `{cohere}_` prefix